### PR TITLE
修复文件延迟访问的问题，加个轮询

### PR DESCRIPTION
**Pull request by yoyo837**
## Summary by CodeRabbit

- **New Features**
	- Enhanced synchronization process with real-time updates on log status.
	- Dynamic button text changes based on synchronization state.

- **Bug Fixes**
	- Improved control flow for showing logs and handling synchronization.

- **Refactor**
	- Updated functions for better clarity and efficiency in handling log states.
	- Introduced a new polling mechanism to check log readiness.

### CODE REVIEW

File: src/components/Sync.tsx [hunk 1]
```diff
@@ -43,9 +70,29 @@ export default function Sync({ pkgName }: SyncProps) {
   return (
     <>
       {contextHolder}
-      <Button size={'small'} type="primary" onClick={logId ? () => showLog(logId) : doSync}>
-        {logId ? '查看日志' : '进行同步'}
+      <Button size={'small'} type="primary" onClick={() => {
```
Comment (elrrrrrrr) [inline src/components/Sync.tsx]: loading = { logState === 1 } ?

**Comment by elrrrrrrr** [comment]
image

可能还要再调整下，默认会显示调度失败，界面看不到日志了 😭

File: src/components/Sync.tsx (hunk 1 above)
Comment (yoyo837) [reply src/components/Sync.tsx]: 已处理

**Comment by yoyo837** [comment]
> image
> 
> 可能还要再调整下，默认会显示调度失败，界面看不到日志了 😭

修改了

File: src/components/Sync.tsx [hunk 2]
```diff
@@ -10,31 +10,59 @@ interface SyncProps {
 
 export default function Sync({ pkgName }: SyncProps) {
   const [logId, setLogId] = React.useState<string>();
+  const [logState, setLogState] = React.useState<number>(1);
+  const retryCountRef = React.useRef(0);
   const [modal, contextHolder] = Modal.useModal();
 
-  async function showLog(id: string) {
+  const logFileUrl = logId ? `${REGISTRY}/-/package/${pkgName}/syncs/${logId}/log`: null;
```
Comment (elrrrrrrr) [inline src/components/Sync.tsx]: logId 是不是异步更新呀？ 看起来还是会卡在等待调度
image

**Review by elrrrrrrr — CHANGES_REQUESTED** [review]

File: src/components/Sync.tsx (hunk 2 above)
Comment (yoyo837) [reply src/components/Sync.tsx]: 调整了

**Comment by elrrrrrrr** [comment]
👍🏻 成功了，弹框内容我后面再优化下

image

**Review by elrrrrrrr — APPROVED** [review]

**Pull request merged**